File: ts/src/actions/SubFlowExecution.ts
import type {ActionSubFlowExecutionResponse} from "@code0-tech/tucana/aquila";
import {CodeZeroEvent} from "../events";
import type {Action} from "../action";

export const packetType = "subFlowExecutionResponse";

export function handle(action: Action, response: ActionSubFlowExecutionResponse): void {
    action.emit(CodeZeroEvent.subFlowExecutionResponseReceived, response);
    action.resolveExecutionResponse(response.executionIdentifier, response.result);
}

File: py/hercules/actions/sub_flow_execution.py
"""Sub flow execution response handler.

Port of ``src/actions/SubFlowExecution.ts``.
"""
from __future__ import annotations

from hercules.events import CodeZeroEvent

packet_type = "sub_flow_execution_response"


def handle(action, response) -> None:
    action.emit(CodeZeroEvent.sub_flow_execution_response_received, response)
    action.resolve_execution_response(response.correlation_identifier, response)
